### ISSUE TITLE
Search_for overwrites custom SELECTs

### ISSUE THREAD

**Issue by cordawyn** [wvanbergen/scoped_search#66]
Hello! Here's my humble addition to the short list of bugs ;-)

Scoped_search seems to overwrite custom `select`s in certain cases.

Take this simple model, for example:

```
class User < ActiveRecord::Base
  has_many :products
  scoped_search :in => :products, :on => :name
end
```
- This works fine: `User.search_for("stuff")`.
- This still works: `User.search_for("stuff").select("count(products.id) as product_count").having("product_count > 0")` (I know that one should better use "where" instead of "having", but it's just a simplified example of the complex query I have locally).
- This doesn't work, because 2 INNER JOINs are generated: `User.search_for("stuff").select("count(products.id) as product_count").joins("INNER JOIN products ON products.user_id = users.id").having("product_count > 0")`, so I decided to go with:
- `User.search_for("stuff").select("count(products.id) as product_count").joins(:products).having("product_count > 0")` - while this helps to avoid  JOIN duplicates, this time `search_for` _replaces_ my custom `select` with its own stuff, without my "product_count", so HAVING statement cannot find "product_count" now.

To summarize, it appears that if I use `joins("INNER JOIN products ON products.user_id = users.id")`, scoped_search does not generate its own SELECT statement. But when I use `joins(:products)`, by specifying the association name instead of hard-coding the SQL statement, scoped_search completely replaces the SELECT statement with that of its own.

I think that instead of trying to be "smart", scoped_search should _always_ use custom `select` whenever that is available.
